util/encoding/encoding.go

```diff
@@ -599,3 +600,57 @@ func DecodeIfNull(b []byte) ([]byte, bool) {
 	}
 	return b, false
 }
+
+// EncodeTime encodes a time value, appends it to the supplied buffer,
+// and returns the final buffer. The encoding is guaranteed to be ordered
+// Such that if t1.Before(t2) then after EncodeTime(b1, t1), and
+// EncodeTime(b2, t1), Compare(b1, b2) < 0. The time zone offset is added
+// as a two byte encoding at the end of the time encoding.
+func EncodeTime(b []byte, t time.Time) []byte {
```

Comment:
Yeah, there aren't guarantees but if we wanted to use it we could check the first byte (version) to make sure that it hasn't changed. I just want to be explicit about the ways in which we deviate from the original (i.e. varint vs fixed-width: how many bytes are we saving for plausible timestamp values?)